### DISCUSSION TITLE
Tika  Max retries exceeded

### DISCUSSION THREAD

**Discussion by Domi31tls** [open-webui/open-webui#4120] [General]
# Bug Report
Tika  Max retries exceeded 
## Description
Config TIKA
![Capture d’écran du 2024-07-25 15-50-01](https://github.com/user-attachments/assets/a99ad9e9-ce9b-4317-b8f3-e904f61ba25a)
**Bug Summary:**
Something went wrong :/ HTTPConnectionPool(host='127.0.0.1', port=9998): Max retries exceeded with url: /tika/text (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x7ab83b219690>: Failed to resolve '127.0.0.1' ([Errno -2] Name or service not known)"))
**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]
When I import my docs
**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment
Tika docker
ubuntu 24.04
Openwebui : v0.3.10

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by EricSchuMa** [comment]
Having the same issue. Were you able to resolve this in the meantime?

**Comment by ruschestor** [comment]
Does Open WebUI run in a container?
If yes, I suppose Open WebUI can't access Tika server because 127.0.0.1 is local address of the container. You should specify host address.
There are two possible opsions:
1. Use direct IP address of the host. Example: http://192.168.1.10:9998
2. "_Use --network="host" in your docker run command, then 127.0.0.1 in your docker container will point to your docker host_"  https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach

**Comment by hidara2000** [comment]
I had the same issue and the following seems to work

Chunk Size
1024
Chunk Overlap
100
``` these were set to 0```

I also had to set a split by token and not character. 
![image](https://github.com/user-attachments/assets/08a7a86d-9efb-420d-9d41-08188f4a7626)

lastly I was getting timeout errors for some uploads and local storage issues for large images.

   client_max_body_size 50M;
   proxy_connect_timeout 300s;
   proxy_send_timeout 300s;
   proxy_read_timeout 300s;

nginx settings
``` config
events {}

http {
  client_max_body_size 50M;
  proxy_connect_timeout 300s;
  proxy_send_timeout 300s;
  proxy_read_timeout 300s;

  server {
      listen 80;
      server_name localhost;
      return 301 https://$host$request_uri;
  }

  server {
      listen 443 ssl;
      server_name localhost;
      ssl_certificate /etc/ssl/certs/cert.pem;
      ssl_certificate_key /etc/ssl/certs/privkey.pem;

      location / {
          proxy_pass http://open-webui:8080;
          proxy_set_header Host $host;
          proxy_set_header X-Real-IP $remote_addr;
          proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
          proxy_set_header X-Forwarded-Proto $scheme;
          proxy_http_version 1.1;
          proxy_set_header Upgrade $http_upgrade;
          proxy_set_header Connection "upgrade";
      }

      location /tika {
          proxy_pass http://tika:9998;
          proxy_set_header Host $host;
          proxy_set_header X-Real-IP $remote_addr;
          proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
          proxy_set_header X-Forwarded-Proto $scheme;
          proxy_buffering off;
      }
  }
}
```

hope it works for someone

**Comment by yousimu** [comment]
I also encountered this issue and once thought that Tika no longer supported `/tika/text`.  
Actually, solving this problem is very simple: you just need to explicitly specify the container's network, for example:

```yaml
networks:
  - openwebui-network

networks:
  openwebui-network:
    driver: bridge
```

I still don’t understand why other services work without specifying the network, but only OpenWebUI doesn’t.

https://github.com/open-webui/open-webui/issues/21119

**Reply by Classic298** [reply]
because other services are not on your localhost.
